extensions/ql-vscode/src/remote-queries/run-remote-query.ts

```diff
@@ -567,14 +567,14 @@ export async function getResultIndex(
   }
 }
 
-function buildRemoteQueryEntity(
+async function buildRemoteQueryEntity(
```

Comment:
Can we prepare the file before calling this function and then just passing the file path?  If we do that then the function's responsibility will remain as is (just builds an entity based on some data), rather than also doing stuff like creating files etc.

We can even put the file writing stuff in a function e.g. `writeQueryTextFile()` and then have `runRemoteQuery()` call that, and then `buildRemoteQueryEntity()`.

Does that make sense?

Reply:
Given that our new approach just passes in the `queryText` now (and no longer writes to a file), I haven't refactored this. Let me know if it's still worth changing though!